data/processSynapseJSON.ts

```diff
@@ -849,6 +875,36 @@ function extractEntitiesFromSynapseData(
                     attributeToId['Dataset Name'] = 'bts:DatasetName';
                 }
 
+                if (synapseRecords.data_schema === 'bts:Therapy') {
+                    synapseRecords.record_list.forEach((record) => {
+                        const therapyEntity: Partial<SerializableEntity> = {
+                            biospecimenIds: [],
+                            diagnosisIds: [],
+                            demographicsIds: [],
+                            therapyIds: [],
+                        };
+
+                        synapseRecords.column_order.forEach((column, i) => {
+                            const id = attributeToId[column];
+
+                            if (id) {
+                                // Ensure TypeScript recognizes therapyEntity properly when indexing
+                                (therapyEntity as any)[
+                                    id.replace(/^bts:/, '')
+                                ] = record.values[i];
+                            }
+                        });
+
+                        therapyEntity.atlasid = atlas.htan_id;
+                        therapyEntity.atlas_name = atlas.htan_name;
+                        therapyEntity.AtlasMeta =
+                            AtlasMetaMap[atlas.htan_id.toUpperCase()];
+                        therapyEntity.Component = 'Therapy';
+                        therapyEntity.assayName = 'Therapy';
+                        therapyEntity.level = 'Therapy';
+                    });
+                }
```

Comment:
We don't probably need to create a new entity, we can handle the therapies as part of `getSampleAndPatientData`. We will need to pass the `therapyByParticipantID` object to that function.